src/segmantic/seg/monai_unet.py

```diff
@@ -548,3 +669,182 @@ def print_table(header, vals, indent="\t"):
+                                    gpu_ids=gpu_ids)
+            elif folder.is_dir() and folder.name == 'Layers[16,32,64,128]':
+                for fold in folder.iterdir():
+                    for file in fold.iterdir():
+                        if file.match('*.ckpt'):
+                            print('start prediction')
+                            predict(model_file=file,
+                                    output_dir=fold,
+                                    test_images=generalize_img,
+                                    test_labels=generalize_label,
+                                    tissue_dict=tissue_dict,
+                                    channels=test_layers[1],
+                                    strides=test_strides[1],
+                                    gpu_ids=gpu_ids)
+            elif folder.is_dir() and folder.name == 'Layers[64,128,256,512,1024]':
+                for fold in folder.iterdir():
+                    for file in fold.iterdir():
+                        if file.match('*.ckpt'):
+                            print('start prediction')
+                            predict(model_file=file,
+                                    output_dir=fold,
+                                    test_images=generalize_img,
+                                    test_labels=generalize_label,
+                                    tissue_dict=tissue_dict,
+                                    channels=test_layers[2],
+                                    strides=test_strides[2],
+                                    gpu_ids=gpu_ids)
+            else:
+                continue
+
+    kf = KFold(n_splits=n_splits)
+
+    image_files = sorted(Path(image_dir).glob("*.nii.gz"))
+    label_files = sorted(Path(labels_dir).glob("*.nii.gz"))
+
+    image_idx = np.arange(len(image_files))
+
+    cv_split = Path(Path(image_dir).parent) / 'CV_Split'
+    cv_split_train = cv_split / 'train'
+    cv_split_predict = cv_split / 'predict'
+    cv_split_train_img = cv_split_train / 'images'
+    cv_split_train_lbl = cv_split_train / 'labels'
+    cv_split_predict_img = cv_split_predict / 'images'
+    cv_split_predict_lbl = cv_split_predict / 'labels'
+
+    all_paths = [cv_split, cv_split_train, cv_split_predict,
+                 cv_split_train_img, cv_split_train_lbl, cv_split_predict_img, cv_split_predict_lbl]
+
+    for path in all_paths:
+        if not path.exists():
+            path.mkdir()
+
+    for config_file in Path(config_files_dir).iterdir():
+        output_dir_scenario = output_dir.joinpath(str(config_file.name))
+        if not output_dir_scenario.exists():
+            output_dir_scenario.mkdir()
+
+        for train_idx, test_idx in kf.split(image_idx, image_idx):
+
+            current_output = output_dir_scenario.joinpath(str(test_idx))
+            print(current_output)
+            if current_output.exists():
+                continue
+            else:
+                current_output.mkdir()
```

Comment:
this will throw an exception if the folder already exists, but maybe that is intended?

Reply:
nope^^ that was just me not knowing about exists_ok when I wrote it ;)